pycalphad/model.py

```diff
@@ -261,6 +261,21 @@ def __init__(self, dbe, comps, phase_name, parameters=None):
         self.site_fractions = sorted([x for x in self.variables if isinstance(x, v.SiteFraction)], key=str)
         self.state_variables = sorted([x for x in self.variables if not isinstance(x, v.SiteFraction)], key=str)
 
+    @staticmethod
+    def unwrap_piecewise(graph):
+        replace_dict = {}
+        for atom in graph.atoms(Piecewise):
+            args = atom.args
+            # Unwrap temperature-dependent piecewise with zero-defaults
+            if len(args) == 4 and args[2] == 0 and args[3] == True and args[1].free_symbols == {v.T}:
+                arg0 = args[0]
+                try:
+                    arg0 = float(arg0)
+                except (ValueError, RuntimeError):
+                    pass
+                replace_dict[atom] = arg0
```

Comment:
What's the use case for coercing for float here? If we fail to coerce to float, the original value is still getting put in the replacement dict. Is that expected behavior?

Reply:
I thought it was necessary to deal with a type promotion issue with non-float constants in SymEngine, but the test suite seems to pass without it (on both symengine.py 0.9.0 and 0.9.2). Not sure what's changed; maybe it was necessary before I moved some code around, and then I forgot to check. In any case, I will remove it now.